File: leetcode/0053-maximum-subarray/solution.py
class Solution(object):
    def maxSubArray(self, nums):
        total=curr_sum=nums[0]

        for num in nums[1:]:
            curr_sum=max(num,curr_sum+num)
            total=max(total,curr_sum)
        return total
        """
        :type nums: List[int]
        :rtype: int
        """
        

File: leetcode/0053-maximum-subarray/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int total,curr_sum;
        total=curr_sum=nums[0];
        nums.erase(nums.begin());
        for (int i:nums){
            curr_sum=max(i,curr_sum+i);
            total=max(total,curr_sum);
        }
        return total;
    }
};
